### ISSUE TITLE
Update CamelCase helper to support abbreviation-prefixed property names

### ISSUE THREAD

**Issue by kamranayub** [frhagn/Typewriter#79]
Given a property name like `IPAddress`, the camel case should be `ipAddress`.

This is the equivalent JSON.NET `ToCamelCase` helper:

```c#
public static string ToCamelCase(string s)
        {
            if (string.IsNullOrEmpty(s))
            {
                return s;
            }

            if (!char.IsUpper(s[0]))
            {
                return s;
            }

            char[] chars = s.ToCharArray();

            for (int i = 0; i < chars.Length; i++)
            {
                bool hasNext = (i + 1 < chars.Length);
                if (i > 0 && hasNext && !char.IsUpper(chars[i + 1]))
                {
                    break;
                }

#if !(DOTNET || PORTABLE)
                chars[i] = char.ToLower(chars[i], CultureInfo.InvariantCulture);
#else
                chars[i] = char.ToLowerInvariant(chars[i]);
#endif
            }

            return new string(chars);
        }
```

**Comment by frhagn** [comment]
Thanks @kamranayub,
I'll change this in the next version

**Comment by kamranayub** [comment]
No problem, as a workaround I can use a custom function :+1:

**Comment by frhagn** [comment]
Resolved in 1.3.1
